core_types/src/entry/agent.rs

```diff
@@ -0,0 +1,13 @@
+use keys::Key;
+
+pub struct Agent {
```

Comment:
I believe this should be renamed `AgentId`, as to not be confused with the module Agent which is the redux state-slice.